File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuProgram.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "su_program",
       indexes = {
           @Index(name = "idx_program_code", columnList = "program_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuProgram extends BaseEntity {

    @Column(name = "parent_program_id", insertable = false, updatable = false)
    private UUID parentProgramId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_program_id")
    private SuProgram parentProgram;

    @Column(name = "program_code", nullable = false, length = 50)
    private String programCode;

    @Column(name = "icon", length = 100)
    private String icon;

    @Column(name = "name_en", nullable = false, length = 255)
    private String nameEn;

    @Column(name = "name_local", nullable = false, length = 255)
    private String nameLocal;

    @Column(name = "route_path", length = 500)
    private String routePath;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "parentProgram", fetch = FetchType.LAZY)
    private List<SuProgram> childPrograms = new ArrayList<>();

    @OneToMany(mappedBy = "program", fetch = FetchType.LAZY)
    private List<SuBusinessRoleProgram> suBusinessRolePrograms = new ArrayList<>();
}

File: sic-api/Entities/Su/SuProgram.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Table("su_program")]
public class SuProgram : BaseEntity
{
    [ForeignKey(nameof(ParentProgram))]
    [Column("parent_program_id")]
    public Guid? ParentProgramId { get; set; }
    public SuProgram? ParentProgram { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("program_code")]
    public string ProgramCode { get; set; } = default!;

    [MaxLength(100)]
    [Column("icon")]
    public string? Icon { get; set; }

    [Required]
    [MaxLength(255)]
    [Column("name_en")]
    public string NameEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("name_local")]
    public string NameLocal { get; set; } = default!;

    [MaxLength(500)]
    [Column("route_path")]
    public string? RoutePath { get; set; }

    [Column("sort_order")]
    public int? SortOrder { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    [Column("role_back")]
    public bool RoleBack { get; set; } = false;

    [Column("role_search")]
    public bool RoleSearch { get; set; } = false;

    [Column("role_add")]
    public bool RoleAdd { get; set; } = false;

    [Column("role_save")]
    public bool RoleSave { get; set; } = false;

    [Column("role_delete")]
    public bool RoleDelete { get; set; } = false;

    [Column("role_print")]
    public bool RolePrint { get; set; } = false;

    public ICollection<SuProgram>? ChildPrograms { get; set; }
    public ICollection<SuBusinessRoleProgram>? SuBusinessRolePrograms { get; set; }
}
